src/corehost/cli/bdl_processor.cpp

```diff
@@ -2,244 +2,328 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#ifdef FEATURE_APPHOST
-
 #include "bdl_processor.h"
 #include "pal.h"
 #include "trace.h"
 #include "utils.h"
 
-void bdl_processor_t::seek(long offset, int origin)
+using namespace bundle;
+
+void bundle_runner_t::seek(long offset, int origin)
 {
-	if (fseek(m_bundle, offset, origin) != 0)
-	{
-		trace::error(_X("Bundle extraction: Seek failure."));
-		throw StatusCode::BundleExtractionIOError;
-	}
+    if (fseek(m_bundle_stream, offset, origin) != 0)
+    {
+        trace::error(_X("Failure processing application bundle; possible file corruption."));
+        trace::error(_X("I/O seek failure within the bundle."));
+        throw StatusCode::BundleExtractionIOError;
+    }
 }
 
-void bdl_processor_t::write(const void* buf, size_t size, FILE *stream)
+void bundle_runner_t::write(const void* buf, size_t size, FILE *stream)
 {
-	if (fwrite(buf, 1, size, stream) != size)
-	{
-		trace::error(_X("Bundle extraction: Write failure."));
-		throw StatusCode::BundleExtractionIOError;
-	}
+    if (fwrite(buf, 1, size, stream) != size)
+    {
+        trace::error(_X("Failure extracting contents of the application bundle."));
+        trace::error(_X("I/O failure when writing extracted files."));
+        throw StatusCode::BundleExtractionIOError;
+    }
 }
 
-void bdl_processor_t::read(void* buf, size_t size, FILE* stream)
+void bundle_runner_t::read(void* buf, size_t size, FILE* stream)
```

Comment:
Do we ever use the read on anything else but the bundle?
There's a bit of an inconsistency here, `seek` doesn't take the stream to operate on (always uses the bundle), `read` does take it, but it feels like it will always be bundle. `write` takes it as well and that one is on the other hand never the bundle.
Maybe make all of these static methods which always take the stream for symmetry. No strong feeling though.
But I think it would make sense to have at least `seek` and `read` consistent.

Reply:
We only `seek()` and `read()` from the bundle file. However, the difference is that while `seek()` is only used within `bundle_runner`, `read()` is used from other types like `manifest` and `file_entry`.  
So, `seek()` was a private member function, while `read()` was a static public function. 

Anyway, I've changed `seek()` now to be a static function taking a stream argument, similar to `read()` and `write().